src/connector/src/sink/kafka.rs

```diff
@@ -491,13 +549,13 @@ impl KafkaSinkWriter {
         Ok(())
     }
 
-    async fn upsert(&self, chunk: StreamChunk) -> Result<()> {
-        let upsert_stream = gen_upsert_message_stream(
-            &self.schema,
-            &self.pk_indices,
-            chunk,
-            UpsertAdapterOpts::default(),
-        );
+    async fn upsert(&mut self, chunk: StreamChunk) -> Result<()> {
+        let schema = self.schema.clone();
```

Comment:
We may add a TODO here for removing these clone in the future. Currently it is unnecessary and it's only for the borrow checker.

Reply:
Sure!